content/docs/testing-environments.md

```diff
@@ -0,0 +1,57 @@
+---
+id: testing-environments
+title: Testing environments
+permalink: docs/testing-environments.html
+---
+
+<!-- This document is intended for folks who are comfortable with Javascript, and have probably written tests with it. It acts as a reference for the differences in testing environments for React components, and how those differences affect the tests that they write. This document also assumes a slant towards web-based react-dom components, but has notes for other renderers. -->
+
+Testing environments can affect the way you write and run tests for React Components. 
+
+This document goes through the major factors that can affect your environment, the implications of those decisions, and recommendations for some scenarios.
+
+### Test frameworks {#test-frameworks}
+
+Test frameworks like [Jest](https://jestjs.io/), [mocha](https://mochajs.org/), [ava](https://github.com/avajs/ava) let you write test suites as regular Javascript, and run them as part of your development process. Additionally, test suites are run as part of workflows like builds, CI/CD, etc.
+
+- Jest is widely compatible with React projects, supporting features like mocked [modules](#mocking-modules) and [timers](#mocking-timers), [jsdom](#mocking-a-rendering-surface) support, etc.
+- Libraries like [mocha](https://mochajs.org/) work well in real browser environments, and could help for tests that explicitly need it.
+- End-to-End tests are used for testing longer flows across multiple pages, and require a [different setup](#end-to-end-tests-aka-e2e-tests)
+
```

Comment:
Maybe add a sentence that suggests how much to test UI components, and when we recommend to use unit / e2e testing.

Reply:
We don't really recommend 'unit' testing in the traditional sense anyway. On L29 I mention how most UI tests are done with a jest/jsdom setup.